template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/IdentityService/Controllers/IdentityServiceController.cs

```diff
@@ -0,0 +1,131 @@
+{
+    [Area(IdentityServiceConstants.Tenant)]
+    [Route("tfp/[area]/" + IdentityServiceConstants.DefaultPolicy + "/oauth2/v" + IdentityServiceConstants.Version + "/[action]")]
+    public class IdentityServiceController : Controller
+    {
+        private readonly IOptions<IdentityServiceOptions> _options;
+        private readonly ITokenManager _tokenManager;
+        private readonly SessionManager<ApplicationUser, IdentityServiceApplication> _sessionManager;
+        private readonly IAuthorizationResponseFactory _authorizationResponseFactory;
+        private readonly ITokenResponseFactory _tokenResponseFactory;
+
+        public IdentityServiceController(
+            IOptions<IdentityServiceOptions> options,
+            ITokenManager tokenManager,
+            SessionManager<ApplicationUser, IdentityServiceApplication> sessionManager,
+            IAuthorizationResponseFactory authorizationResponseFactory,
+            ITokenResponseFactory tokenResponseFactory)
+        {
+            _options = options;
+            _tokenManager = tokenManager;
+            _sessionManager = sessionManager;
+            _authorizationResponseFactory = authorizationResponseFactory;
+            _tokenResponseFactory = tokenResponseFactory;
+        }
+
+        [AcceptVerbs("GET", "POST")]
+        [ActionName("Authorize")]
+        public async Task<IActionResult> AuthorizeAsync(
+            [EnableIntegratedWebClient, ModelBinder(typeof(AuthorizationRequestModelBinder))] AuthorizationRequest authorization)
+        {
+            if (!authorization.IsValid)
+            {
+                return this.InvalidAuthorization(authorization.Error);
+            }
+
+            var authorizationResult = await _sessionManager.IsAuthorizedAsync(authorization);
+            if (authorizationResult.Status == AuthorizationStatus.Forbidden)
+            {
+                return this.InvalidAuthorization(authorizationResult.Error);
+            }
+
+            if (authorizationResult.Status == AuthorizationStatus.LoginRequired)
+            {
+                return RedirectToLogin(nameof(AccountController.Login), "Account", authorization.Message);
+            }
+
+            var context = authorization.CreateTokenGeneratingContext(
+                authorizationResult.User,
+                authorizationResult.Application);
+
+            context.AmbientClaims.Add(new Claim("policy", IdentityServiceConstants.DefaultPolicy));
+            context.AmbientClaims.Add(new Claim("version", "1.0"));
+            context.AmbientClaims.Add(new Claim("tenantId", IdentityServiceConstants.TenantId));
+
+            await _tokenManager.IssueTokensAsync(context);
+            var response = await _authorizationResponseFactory.CreateAuthorizationResponseAsync(context);
+
+            await _sessionManager.StartSessionAsync(authorizationResult.User, authorizationResult.Application);
+
+            return this.ValidAuthorization(response);
```

Comment:
Actually, it's required in this case because its an identity service specific extension method, not a helper on the base class.